### ISSUE TITLE
Enable Data Retention / Expiry to be "dynamic" based on latest interaction 

### ISSUE THREAD

**Issue by fpietrosanti** [globaleaks/GlobaLeaks#1775]
This ticket is to enable Data Retention / Expiry to be "dynamic" based on latest interaction of Receiver or Whistleblower.

Actually the data retention of Tip is static (if i set 200 days, after 200 days the specific Tip sent 200 days ago will be deleted).

TI-it (Giorgio) asked this to become also a dynamic setting, so that the amount of days after which a Tip is automatically deleted/cleaned up is based on the  latest interaction of Receiver or Whistleblower. 

The assumption is that if neither the Receiver neither the Whistleblower accessed the Tip in the latests X days, then the Tip is not to be considered anymore useful.

**Comment by NSkelsey** [comment]
There is a button which a receiver can press that delays the expiration of the tip. Also access expires for the whistleblower after a certain amount of time if they do not visit the tip.

I do not think it is a good idea to add the same access based expiration functionality for the receiver because it means that just by looking at a tip again I push its expiration further out. The interface should be explicit where if I want to keep data around I can press a button and the tip stays.

We can add a switch to this, but this is a low priority. Right now you have to visit the tip and also click a button. The functionality change saves the recipient a click. And actually there is a mass delay tip expiration selector in the tip list which you can use to preserve the tips that you want.

IMHO this is a bad suggestion.

**Comment by fpietrosanti** [comment]
@NSkelsey yes but that's the reality of their operations, they have tip where they work for very long time and tip that they stop using quickly. So the security vs. flexibility tradeoff is to have also a dynamic data retention policy, compared to the one actually statically configured.

**Comment by NSkelsey** [comment]
There is also a delete button.

**Comment by fpietrosanti** [comment]
@NSkelsey yes, but this imply a manual handling by the person working on the whistleblowing system. 
Data retention policy is designed to keep the system clean, after a tip is not useful anymore deleting it. 

Up to now it has been a "static settings", but this experience suggest to introduce a "dynamic setting" for better determining "when a tip is not useful anymore and can be deleted" .

In future there will be also very advanced data retention policies based on the content of the Tip (we've conversation with the privacy obdusman on defining a regulatory framework on which kind of data of a whistleblowing system should be preserved and which not, in which condition, on the EU privacy legal basis, but that's still a chat, not yet anything defined).

**Comment by fpietrosanti** [comment]
(btw it's not a urgent implementation, it's just a requirement collected from a user)

**Comment by NSkelsey** [comment]
Okay, I see the vision of clearly defining data-retention policies to be applied to submissions.

A trained compliance officer has a list of options to classify a submission. From those options they can choose a policy for the submission, where the specifies how data retention will be handled.

A set of policies (maybe defined by international bodies) could look like:

System default:
  - whistleblower access expires in 15 days
  - data expires after 30 days

WIN Data retention Reccomendation:
  - whistleblower access expires after 30 days
  - default data expiration of 50 days
  - data expires after inactivity of 10 days
  - Recipient can choose to preserve indefinitely

Hermes Center HR Activist Privacy Reccomendation:
  - XXX
  - XXX
  - XXX

**Comment by fpietrosanti** [comment]
@evilaliv3 was this already done, right?

**Comment by evilaliv3** [comment]
Nope @fpietrosanti its still a todo

**Comment by dmalacaria** [comment]
Hi, I have high interest in this enhancement. What has been already done? Other than have different access expiration for whistleblower and recipient, it could be useful to set Tip data retention based on Tip status, something like: delete after X days it is "closed".
